packages/@aws-cdk/aws-stepfunctions/lib/input.ts

```diff
@@ -3,7 +3,7 @@ import { JsonPath } from './fields';
 /**
  * Type union for task classes that accept multiple types of payload
  */
-export class TaskInput {
+export class TaskInput<ValueType=string | { [key: string]: any; }> {
```

Comment:
I'm not entirely sure that this will this work for all languages that we support. the constructor is private so the static APIs are stable but can we verify?

have you tried running `yarn package` and verified that the generated clients for Python, .NET, Java?

Reply:
I have not - will try this and report back.

Reply:
I'll also try to take this change down and try to get this one shipped sooner than later.
I think I grasp the problem you're trying to solve a lot clearer now and just want to be cautious that we don't break any of the other language support in getting there 🙂

Reply:
Makes sense, thanks @shivlaks, I'll try and make the time for this tonight. 👍🏻